src/ValueHandler/ChannelPricingValueHandler.php

```diff
@@ -28,16 +30,36 @@ final class ChannelPricingValueHandler implements ValueHandlerInterface
     /** @var string */
     private $akeneoAttribute;
 
+    /** @var string */
+    private $syliusPropertyPath;
+
+    /** @var PropertyAccessorInterface */
+    private $propertyAccessor;
+
     public function __construct(
         FactoryInterface $channelPricingFactory,
         ChannelRepositoryInterface $channelRepository,
         RepositoryInterface $currencyRepository,
-        string $akeneoAttribute
+        PropertyAccessorInterface $propertyAccessor = null,
+        string $akeneoAttribute,
+        string $syliusPropertyPath = 'price'
     ) {
         $this->channelPricingFactory = $channelPricingFactory;
         $this->channelRepository = $channelRepository;
         $this->currencyRepository = $currencyRepository;
         $this->akeneoAttribute = $akeneoAttribute;
+        $this->syliusPropertyPath = $syliusPropertyPath;
+        if ($propertyAccessor === null) {
+            trigger_deprecation(
+                'webgriffe/sylius-akeneo-plugin',
+                '1.12',
+                'Not passing a property accessor to "%s" is deprecated and will be removed in %s.',
+                __CLASS__,
+                '2.0'
+            );
+            $propertyAccessor = PropertyAccess::createPropertyAccessor();
+        }
+        $this->propertyAccessor = $propertyAccessor;
```

Comment:
Not passing the `propertyAccessor` should lead to old behavior which sets the price.
```suggestion
        if ($propertyAccessor === null) {
            trigger_deprecation(
                'webgriffe/sylius-akeneo-plugin',
                '1.12',
                'Not passing a property accessor to "%s" is deprecated and will be removed in %s.',
                __CLASS__,
                '2.0'
            );
        }
        $this->syliusPropertyPath = $syliusPropertyPath;
```

Reply:
There is a reason for why not instantiate statically the` $propertyAccessor` with `PropertyAccess::createPropertyAccessor();`

Reply:
Because we'll have to support also the case where the property accessor is statically created. I'd like to support only two cases: property accessor is passed or not.